controllers/barbican_controller.go

```diff
@@ -554,6 +571,42 @@ func (r *BarbicanReconciler) workerDeploymentCreateOrUpdate(ctx context.Context,
 	return deployment, op, err
 }
 
+func (r *BarbicanReconciler) keystoneListenerDeploymentCreateOrUpdate(ctx context.Context, instance *barbicanv1beta1.Barbican, helper *helper.Helper) (*barbicanv1beta1.BarbicanKeystoneListener, controllerutil.OperationResult, error) {
+
+	r.Log.Info(fmt.Sprintf("Creating barbican KeystoneListener spec.  transporturlsecret: '%s'", instance.Status.TransportURLSecret))
+	r.Log.Info(fmt.Sprintf("database hostname: '%s'", instance.Status.DatabaseHostname))
+	keystoneListenerSpec := barbicanv1beta1.BarbicanKeystoneListenerSpec{
+		BarbicanTemplate:                 instance.Spec.BarbicanTemplate,
+		BarbicanKeystoneListenerTemplate: instance.Spec.BarbicanKeystoneListener,
+		DatabaseHostname:                 instance.Status.DatabaseHostname,
+		TransportURLSecret:               instance.Status.TransportURLSecret,
+	}
+
+	deployment := &barbicanv1beta1.BarbicanKeystoneListener{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-keystone-listener", instance.Name),
+			Namespace: instance.Namespace,
+		},
+	}
+
+	op, err := controllerutil.CreateOrUpdate(ctx, r.Client, deployment, func() error {
+		r.Log.Info("Setting deployment spec to be keystonelistenerspec")
+		deployment.Spec = keystoneListenerSpec
+
+		err := controllerutil.SetControllerReference(instance, deployment, r.Scheme)
+		if err != nil {
+			return err
+		}
+
+		// Add a finalizer to prevent user from manually removing child BarbicanAPI
+		controllerutil.AddFinalizer(deployment, helper.GetFinalizer())
```

Comment:
Since we're adding a finalizer to the `BarbicanKeystoneListener` resource, we will also need to remove the finalizer in our deletion logic.  i.e. like we do here for the `BarbicanAPI` resource that this `Barbican` resource owns: https://github.com/openstack-k8s-operators/barbican-operator/blob/ed7f9a3d2080355694c067940194917f5f3c3d7f/controllers/barbican_controller.go#L367-L382